configure.ac

```diff
@@ -526,13 +518,17 @@ PRETTY_ARG_VAR([WARNING_CFLAGS], [C compiler warning flags],
 
 # Checks for library functions
 # ----------------------------
-AC_CHECK_FUNCS(fnmatch,[],[
+AC_CHECK_FUNCS(fnmatch,[have_fnmatch=yes],[
 AC_LIBOBJ([fnmatch])
-CPPFLAGS="$CPPFLAGS -I${srcdir}/fnmatch"
-AC_DEFINE(HAVE_FNMATCH)
-])
+have_fnmatch=no])
 AC_CHECK_HEADERS(fnmatch.h)
 
+AM_CONDITIONAL([HAVE_FNMATCH], [test "x$have_fnmatch" = "xyes"])
+if test "xno" = "x$have_fnmatch"; then
+    CPPFLAGS="$CPPFLAGS -I${srcdir}/fnmatch"
```

Comment:
this should be done in the Makefile, and not touch the user variable `CPPFLAGS` (as it is, if the user sets `CPPFLAGS` when invoking Make (which is totally valid and expected to be supported by Autotools' standards), it'll fail miserably if it requires that extra inclusion but the user didn't know)